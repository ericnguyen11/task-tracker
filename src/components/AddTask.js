import { useState } from "react";

const AddTask = function ({ onAdd }) {
  // Initiate state
  const [text, setText] = useState("");
  const [date, setDate] = useState("");
  const [reminder, setReminder] = useState(false);

  // On submit, alert the user if there is no task test. Otherwise, send task data to onAdd in App.js and reset state
  const onSubmit = function (e) {
    e.preventDefault();

    if (!text) {
      alert("Please add a task");
      return;
    }

    onAdd({ text, date, reminder });

    setText("");
    setDate("");
    setReminder(false);
  };

  return (
    <form className="add-form" onSubmit={onSubmit}>
      <div className="form-layout">
        <label>Task</label>
        <input
          type="text"
          placeholder="Add task"
          value={text}
          onChange={(e) => setText(e.target.value)}
        ></input>
      </div>
      <div className="form-layout">
        <label>Date</label>
        <input
          type="text"
          placeholder="Add date"
          value={date}
          onChange={(e) => setDate(e.target.value)}
        ></input>
      </div>
      <div className="form-layout form-layout-check">
        <label>Set Reminder</label>
        <input
          type="checkbox"
          checked={reminder}
          value={reminder}
          onChange={(e) => setReminder(e.currentTarget.checked)}
        />
      </div>
      <input type="submit" value="Save Task" className="btn btn-save" />
    </form>
  );
};

export default AddTask;
