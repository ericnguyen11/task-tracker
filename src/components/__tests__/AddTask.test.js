import { shallow } from "enzyme";

import AddTask from "../AddTask.js";

it("renders without crashing", () => {
  shallow(<AddTask />);
});

it("matches snapshot", () => {
  const wrapper = shallow(<AddTask />);
  expect(wrapper).toMatchSnapshot();
});
