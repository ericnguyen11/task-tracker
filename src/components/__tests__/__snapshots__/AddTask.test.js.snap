// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<form
  className="add-form"
  onSubmit={[Function]}
>
  <div
    className="form-layout"
  >
    <label>
      Task
    </label>
    <input
      onChange={[Function]}
      placeholder="Add task"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-layout"
  >
    <label>
      Date
    </label>
    <input
      onChange={[Function]}
      placeholder="Add date"
      type="text"
      value=""
    />
  </div>
  <div
    className="form-layout form-layout-check"
  >
    <label>
      Set Reminder
    </label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
      value={false}
    />
  </div>
  <input
    className="btn btn-save"
    type="submit"
    value="Save Task"
  />
</form>
`;
